### DISCUSSION TITLE
Probably obvious but, how can I define the 'user config directory'.

### DISCUSSION THREAD

**Discussion by baudneo** [ultralytics/ultralytics#4045] [Q&A]
As an example, I am using yolov8 to infer on objects. I have a venv for the deps and package. I receive this message every time the server component is run:

```
WARNING ⚠️ user config directory is not writeable, defaulting to '/tmp/Ultralytics'.
```

In the docs, the only way to configure a writable directory is during training using MODE(ARGS) with `project=` and `name=`. How can I configure this while using detect+predict ?

TIA.

**Comment by glenn-jocher** [comment]
@baudneo hello! The warning you're seeing indicates that the default user config directory isn't writeable, and as a result, the system is defaulting to using `/tmp/Ultralytics`. To configure a writable directory for your use case, you can set the `YOLOv8_DIR` environment variable to a path where you have write permissions before running your server component.

For example, in your virtual environment, you can export the `YOLOv8_DIR` environment variable like this:

```sh
export YOLOv8_DIR=/path/to/your/writable/directory
```

Make sure to replace `/path/to/your/writable/directory` with the actual path you want to use. Once set, YOLOv8 will use this directory for storing user configurations and other necessary files.

If you're running this in a script or a server environment, ensure that the environment variable is set in the context where the server is running.

For more details on configuration and environment variables, you can refer to the documentation at https://docs.ultralytics.com. 😊

**Reply by baudneo** [reply]
Hi, thanks for the advice @glenn-jocher !

**Reply by glenn-jocher** [reply]
@baudneo you're welcome! If you have any more questions or need further assistance, feel free to reach out. Good luck with your object detection tasks using YOLOv8! 😄🚀
